### PR TITLE
PEP8 compliance

### PR DESCRIPTION
**Pull request by matthewrmshin**
Modify source to be strictly PEP8 compliance.
Add PEP8 test.

Address one aspect of #129.

### CODE REVIEW

**Comment by matthewrmshin** [comment]
@benfitzpatrick @arjclark please review.

**Comment by benfitzpatrick** [comment]
Conflicts now...

File: lib/python/rose/apps/comparisons/output_grepper.py [hunk 1]
```diff
@@ -16,29 +16,30 @@
...
 import re
 
 REGEXPS = {
-  'um_wallclock' : r"Maximum Elapsed Wallclock Time:\s*(\S+)",
-  'um_initial_norms' : r"initial\s*Absolute\s*Norm\s*:\s*(\S+)",
-  'um_final_norms' : r"Final\s*Absolute\s*Norm\s*:\s*(\S+)", }
+    'um_wallclock': r"Maximum Elapsed Wallclock Time:\s*(\S+)",
+    'um_initial_norms': r"initial\s*Absolute\s*Norm\s*:\s*(\S+)",
+    'um_final_norms': r"Final\s*Absolute\s*Norm\s*:\s*(\S+)"}
+
```
Comment (benfitzpatrick) [inline lib/python/rose/apps/comparisons/output_grepper.py]: Probably prefer a closing brace here, but that's personal preference.

File: lib/python/rose/config_editor/variable.py [hunk 2]
```diff
@@ -216,7 +216,7 @@ def handle_bad_valuewidget(self, error_info, variable, set_value):
     def handle_focus_in(self, widget, event):
         widget._first_colour = widget.style.base[gtk.STATE_NORMAL]
         new_colour = rose.gtk.util.color_parse(
-                         rose.config_editor.COLOUR_VALUEWIDGET_BASE_SELECTED)
```
Comment (benfitzpatrick) [inline lib/python/rose/config_editor/variable.py]: Looks fine up to here... to be continued...

**Comment by matthewrmshin** [comment]
Conflicts resolved.

**Comment by benfitzpatrick** [comment]
Looks fine, over to @arjclark.

File: lib/python/rose/apps/comparisons/output_grepper.py (hunk 1 above)
Comment (arjclark) [inline lib/python/rose/apps/comparisons/output_grepper.py]: It'd certainly make the above more readable

File: lib/python/rose/apps/rose_arch_compressions/rose_arch_gzip.py [hunk 3]
```diff
@@ -41,11 +41,11 @@ def compress_sources(self, target, work_dir):
         """
         for source in target.sources.values():
             if source.path.endswith("." + target.compress_scheme):
-                continue # assume already done
+                continue  # assume already done
```
Comment (arjclark) [inline lib/python/rose/apps/rose_arch_compressions/rose_arch_gzip.py]: Not sure why you've had to add extra whitespace here? Seems surplus

File: lib/python/rose/apps/rose_arch_compressions/rose_arch_tar.py [hunk 4]
```diff
@@ -44,9 +44,9 @@ def compress_sources(self, target, work_dir):
         sources = target.sources.values()
         scheme = target.compress_scheme
         if (len(sources) == 1 and
-            sources[0].path.endswith("." + target.compress_scheme)):
+                sources[0].path.endswith("." + target.compress_scheme)):
             target.work_source_path = sources[0].path
-            return # Assume that it has been done
+            return  # Assume that it has been done
```
Comment (arjclark) [inline lib/python/rose/apps/rose_arch_compressions/rose_arch_tar.py]: As with above extra whitespace query

File: lib/python/rose/config_editor/keywidget.py [hunk 5]
```diff
@@ -37,25 +37,18 @@ class KeyWidget(gtk.VBox):
...
-             rose.config_editor.FLAG_TYPE_OPTIONAL:
-                  gtk.STOCK_ABOUT,
-             rose.config_editor.FLAG_TYPE_NO_META:
-                  gtk.STOCK_DIALOG_QUESTION}
+        rose.config_editor.FLAG_TYPE_DEFAULT: gtk.STOCK_INFO,
+        rose.config_editor.FLAG_TYPE_ERROR: gtk.STOCK_DIALOG_WARNING,
+        rose.config_editor.FLAG_TYPE_FIXED: gtk.STOCK_DIALOG_AUTHENTICATION,
+        rose.config_editor.FLAG_TYPE_OPT_CONF: gtk.STOCK_INDEX,
+        rose.config_editor.FLAG_TYPE_OPTIONAL: gtk.STOCK_ABOUT,
+        rose.config_editor.FLAG_TYPE_NO_META: gtk.STOCK_DIALOG_QUESTION}
```
Comment (arjclark) [inline lib/python/rose/config_editor/keywidget.py]: As with Ben's above comment, this'd be more readable and make more sense if the final brace was on a separate line.

**Comment by arjclark** [comment]
Looks ok to me and test-battery passes in my environment. A couple of style choice queries above.

File: lib/python/rose/apps/rose_arch_compressions/rose_arch_gzip.py (hunk 3 above)
Comment (matthewrmshin) [inline lib/python/rose/apps/rose_arch_compressions/rose_arch_gzip.py]: PEP8 demands it.

File: lib/python/rose/apps/rose_arch_compressions/rose_arch_tar.py (hunk 4 above)
Comment (matthewrmshin) [inline lib/python/rose/apps/rose_arch_compressions/rose_arch_tar.py]: PEP8 demands it.

File: lib/python/rose/apps/comparisons/output_grepper.py (hunk 1 above)
Comment (matthewrmshin) [inline lib/python/rose/apps/comparisons/output_grepper.py]: I am surprised you would both say this, given that closing-brace-on-separate-line was my old preferred style, borrowed from languages that demand open and close braces for blocks. It is certainly a personal taste in Python code, when blocks are done by indentation. I agree that closing-brace-on-same style is probably less readable, but I think it looks prettier in Python code.

PEP8 does not appear to complain either way.

**Comment by matthewrmshin** [comment]
I have modified the style of the 2 blocks to that preferred by the team.

**Comment by arjclark** [comment]
Looks good to me.

**Pull request merged**